app/controllers/schools/partnerships_controller.rb

```diff
@@ -6,7 +6,15 @@ class Schools::PartnershipsController < Schools::BaseController
 
   def index
     @school = current_user.induction_coordinator_profile.schools.first
-    @partnership = @school.partnerships.find_by(cohort: cohort)
+    @partnership = @school.partnerships.unchallenged.find_by(cohort: cohort)
+
+    if @partnership
+      email = @partnership.partnership_notification_emails.order(:created_at).first
+      if email.present? && !email.token_expired?
+        @report_mistake_link = challenge_partnership_path(token: email.token)
+        @mistake_link_expiry = email.token_expiry.strftime("%d/%m/%Y")
+      end
+    end
```

Comment:
I'm probably too late here - I really don't think going back to the email record is a good option here. Email represents, well, an email that has been sent and should not take a responsibility of deciding the deadline for partnership challange - this is strictly partnership model responsibility. Token is only used as a mean of authorisation and should not be required for a signed-in user.